### PR TITLE
1117 add new run parameters

### LINKED ISSUE #1117: Add TestPlanRun / TestStepRun property: Status
Currently, there is no reliable way to check if a teststep or testplan ran to completion, was aborted due to a break condition, or was aborted by the user. I think we should add a public status Enum to stepruns / planruns indicating which of the following is the case:

1. Finished normally
2. Was cancelled by the operator
3. Was cancelled due to a break condition

This status should be written to results etc.

### PR DESCRIPTION
**Pull request by alnlarsen**
This adds some run parameters to provide additional details about a test plan run

1. CancelledDueToBreakCondition to indicate whether a test plan run finished normally, or was terminated early due to a break condition being satisfied
2. StepListOverridden to indicate that only a portion of the test plan was run

Closes #1117

### CODE REVIEW

File: Engine/TestPlanExecution.cs [hunk 1]
```diff
@@ -589,6 +599,8 @@ private TestPlanRun DoExecute(IEnumerable<IResultListener> resultListeners, IEnu
             else
             {
                 execStage = new TestPlanRun(this, resultListeners.ToList(), initTime, initTimeStamp);
+                if (stepsOverride != null)
+                    execStage.Parameters.Add(new ResultParameter("StepListOverridden", "Yes"));
```
Comment (rmadsen-ks) [inline Engine/TestPlanExecution.cs:603]: Maybe this should be the list of guids in stepsOverride?

File: Engine/TestPlanExecution.cs [hunk 2]
```diff
@@ -180,6 +180,14 @@ failState execTestPlan(TestPlanRun execStage, IList<ITestStep> steps)
             Stopwatch planRunOnlyTimer = Stopwatch.StartNew();
             var runs = new List<TestStepRun>();
 
+            bool didBreak = false;
+            void addBreakResult()
+            {
+                if (didBreak) return;
+                didBreak = true;
+                execStage.Parameters.Add(new ResultParameter("CancelledDueToBreakCondition", "Yes"));
```
Comment (rmadsen-ks) [inline Engine/TestPlanExecution.cs:188]: Lets move this property name to a static and internal variable in TestPlanRun. Also we could use the GUID of the step causing the plan to cancel?

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
Looks good, but I had minor suggestions for improvements.

File: Engine/TestPlanExecution.cs (hunk 1 above)
Comment (alnlarsen) [reply Engine/TestPlanExecution.cs]: I agree!

File: Engine/TestPlanExecution.cs (hunk 2 above)
Comment (alnlarsen) [reply Engine/TestPlanExecution.cs]: Good call!

File: Engine/TestPlanExecution.cs [hunk 3]
```diff
@@ -701,8 +720,10 @@ public TestPlanRun Execute(IEnumerable<IResultListener> resultListeners, IEnumer
         }
 
         TestPlanRun currentExecutionState = null;
+        private const string breakConditionParameterName = "BreakIssuedFrom";
```
Comment (rmadsen-ks) [inline Engine/TestPlanExecution.cs:723]: If you make these internal, we can use them in the unit tests too.

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]

File: Engine/TestPlanExecution.cs [hunk 4]
```diff
@@ -701,8 +720,10 @@ public TestPlanRun Execute(IEnumerable<IResultListener> resultListeners, IEnumer
         }
 
         TestPlanRun currentExecutionState = null;
+        internal const string breakConditionParameterName = "BreakIssuedFrom";
```
Comment (rmadsen-ks) [inline Engine/TestPlanExecution.cs:723]: Lets move these to a class called SpecialParameterNames. And make them public.

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
Lets make the names more easilty discoverable.

Also should we use spaces?

File: Engine/TestPlanExecution.cs (hunk 4 above)
Comment (alnlarsen) [reply Engine/TestPlanExecution.cs]: Great idea. I have added it.

**Review by rmadsen-ks — APPROVED** [review]
Looks great!

**Pull request merged**